File: python/geometricSum/model/metamorphic_group.py
class MetamorphicGroup:
    """
    表示蜕变测试中的测试组，包含一个源测试用例和一个后续测试用例

    该类保存蜕变关系的信息以及相关的测试用例

    Attributes:
        mr_id (str): 蜕变关系ID (例如, "mr_1")
        description (str): 蜕变关系的描述
        source_test (TestCase): 源测试用例
        followup_test (TestCase): 后续测试用例
    """

    def __init__(self, mr_id, description, source_test, followup_test):
        """
        创建一个新的蜕变测试组

        Args:
            mr_id (str): 蜕变关系ID
            description (str): 蜕变关系的描述
            source_test (TestCase): 源测试用例
            followup_test (TestCase): 后续测试用例
        """
        self.mr_id = mr_id
        self.description = description
        self.source_test = source_test
        self.followup_test = followup_test

    def get_mr_id(self):
        """获取蜕变关系ID"""
        return self.mr_id

    def get_description(self):
        """获取蜕变关系描述"""
        return self.description

    def get_source_test(self):
        """获取源测试用例"""
        return self.source_test

    def get_followup_test(self):
        """获取后续测试用例"""
        return self.followup_test

    def __str__(self):
        """
        返回蜕变测试组的字符串表示

        Returns:
            str: 格式化的蜕变测试组字符串
        """
        return (
            f"MetamorphicGroup[{self.mr_id}]: {self.description}\n"
            f"  Source: {self.source_test}\n"
            f"  Followup: {self.followup_test}"
        )

    def __repr__(self):
        """
        返回蜕变测试组的代码表示形式

        Returns:
            str: 可用于重建对象的字符串表示
        """
        return (
            f"MetamorphicGroup('{self.mr_id}', '{self.description}', "
            f"{repr(self.source_test)}, {repr(self.followup_test)})"
        )

File: java/src/main/java/paper/pss/exp/math2_project/model/MetamorphicGroup.java
package paper.pss.exp.math2_project.model;

/**
 * MetamorphicGroup.java
 * 
 * This file defines the data structure for metamorphic groups used in
 * metamorphic testing of the copySign function.
 * A metamorphic group represents a pair of test cases: a source test case and a
 * follow-up test case, along with the identifier of the metamorphic relation they're testing.
 * 
 * In this simplified model:
 * - Each MetamorphicGroup contains exactly one source test and one follow-up test
 * - Multiple follow-up tests for the same source require multiple MetamorphicGroup instances
 * - Verification logic is handled by the metamorphic relation classes, not in this data structure
 */
public class MetamorphicGroup {
    private final String mrId; // Metamorphic relation ID (e.g., "MR1")
    private final String description; // Brief description of the relation
    private final TestCase sourceTest; // Original test case
    private final TestCase followupTest; // Derived follow-up test case

    /**
     * Creates a metamorphic group with a source and follow-up test case
     * 
     * @param mrId         Metamorphic relation ID (e.g., "MR1")
     * @param description  Brief description of the relation
     * @param sourceTest   Source test case
     * @param followupTest Follow-up test case
     */
    public MetamorphicGroup(String mrId, String description, TestCase sourceTest, TestCase followupTest) {
        this.mrId = mrId;
        this.description = description;
        this.sourceTest = sourceTest;
        this.followupTest = followupTest;
    }

    /**
     * Gets the metamorphic relation ID
     * 
     * @return Relation ID (e.g., "MR1")
     */
    public String getMRId() {
        return mrId;
    }

    /**
     * Gets the description of the metamorphic relation
     * 
     * @return Description text
     */
    public String getDescription() {
        return description;
    }

    /**
     * Gets the source test case
     * 
     * @return Source test case
     */
    public TestCase getSourceTest() {
        return sourceTest;
    }

    /**
     * Gets the follow-up test case
     * 
     * @return Follow-up test case
     */
    public TestCase getFollowupTest() {
        return followupTest;
    }

    /**
     * Returns a string representation of the metamorphic group
     * 
     * @return String representation
     */
    @Override
    public String toString() {
        return String.format("MetamorphicGroup[%s]: %s -> %s", 
                mrId, sourceTest.toString(), followupTest.toString());
    }
}